lib/ansible/module_utils/urls.py

```diff
@@ -1399,6 +1399,18 @@ def fetch_url(module, url, data=None, headers=None, method=None,
         try:
             # Lowercase keys, to conform to py2 behavior, so that py3 and py2 are predictable
             info.update(dict((k.lower(), v) for k, v in e.info().items()))
+            # Don't be lossy, append header values for duplicate headers
+            # In Py2 there is nothing that needs done, py2 does this for us
+            if PY3:
+                temp_headers = {}
```

Comment:
Let's proceed with moving this into a common function so that we don't duplicate this code.

Something like: `normalize_headers`

Then we can do:

```
info.update(normalize_headers(r.headers))
```

```
info.update(normalize_headers(e.headers))
```

In fact, I already have an implementation of this available at https://github.com/ansible/ansible/pull/41925/files#diff-3448ba8582852f79eb0d697763e046d8R880

So I'd recommend just copying that function into this PR, and utilizing it from both locations.

Reply:
Yep great idea, just push the modifications